### ISSUE TITLE
Testing from command line

### ISSUE THREAD

**Issue by JulienPalard** [pappasam/jedi-language-server#102]
According to the README I should be able to test from command line, so I grabbed a json (from strace again):

```json
{
  "jsonrpc": "2.0",
  "method": "textDocument/documentHighlight",
  "params": {
    "textDocument": {
      "uri": "file:///home/mdk/clones/jedi-language-server/test.py"
    },
    "position": {
      "line": 1,
      "character": 4
    }
  },
  "id": 8
}
```

and trying to send it to jedi-language-server using `jedi-language-server < request.json`, but it does *not* return, it looks to fall in an infinite loop of `stats`, (like monitoring my Python files).

**Comment by pappasam** [comment]
Yep, it definitely doesn't work correctly at this time. Thanks for raising, I'll look into this sometime this week, might be helpful to look into a new CLI option about logging or something: https://github.com/openlawlibrary/pygls/blob/470f88f2bcb046781c41077fef921c758e044995/docs/source/pages/advanced_usage.rst#stdio

**Comment by siegel** [comment]
Many of the servers I've been testing with will log to `stderr` which is quite convenient for debugging. The `initialize` parameters have a `trace` key/value (one of `{off, message, verbose}`) that might be useful; in addition some servers support a `--verbose` option on the command line. Happy to write up a new issue if that would help.

**Comment by pappasam** [comment]
Amazing. When I find time, I'll:

1. Migrate to `argparse` to remove the click dependency
2. Add tcp support for command line exploration. I'm guessing that'll be more straightforward to test out interactively without an editor
3. Add a verbose option to log to standard error
4. Make the log level configurable at the command line as well
5. Add another option to write logs to a file on disk (which some people might prefer). Will be generally helpful for future bug reports

**Comment by pappasam** [comment]
Resolved in version `0.29.0`
